producer/src/contract/contract_test.go

```diff
@@ -136,184 +137,230 @@ func TestYieldSerialization(t *testing.T) {
+	pubKey := generatePubKey()
+	yield := MakeYield(&pubKey, amt)
+	contractHash := block.HashSHA256([]byte("blokchain"))
+	blockHeight := uint64(35)
+	yieldIndex := uint16(1)
+
+	err := InsertYield(yield, db, blockHeight, contractHash, yieldIndex)
 	if err != nil {
 		t.Errorf("Failed to insert yield")
 	}
-	testClaim, err := MakeClaim("testDB.dat", testPubKey, 10000000)
+
+	testClaim, err := MakeClaim(db, pubKey, amt)
 	if err != nil {
-		t.Errorf("Failed to claim yield")
+		t.Errorf("Failed to claim yield: %v", err)
 	}
+
 	if !bytes.Equal(testClaim.PreviousContractHash, contractHash) {
 		t.Errorf("Contract hashes do not match")
 	}
-	if testClaim.BlockIndex != 35 {
+	if testClaim.BlockIndex != blockHeight {
 		t.Errorf("Block indeces do not match")
 	}
-	if testClaim.YieldIndex != 1 {
+	if testClaim.YieldIndex != yieldIndex {
 		t.Errorf("Yield indeces do not match")
 	}
-	if !cmp.Equal(testClaim.PublicKey, testPubkey) {
+	if !reflect.DeepEqual(testClaim.PublicKey, pubKey) {
 		t.Errorf("Public Keys do not match")
 	}
 	// Case 3
-	_, shouldNotBeNil := MakeClaim("testDB.dat", testPubKey, 1)
+	_, shouldNotBeNil := MakeClaim(db, pubKey, 1)
 	if shouldNotBeNil == nil {
 		t.Errorf("Made claim on empty yield pool")
 	}
 }
 
-func TestMakeClaimCase1B(t *testing.T) {
-	setUpDB("testDB.db")
-	defer tearDown("testDB.db")
-	testPubKey := generatePubKey()
-	testYield := MakeYield(testPubKey, 50000)
-	contractHash := block.HashSHA256([]byte{'b', 'l', 'k', 'c', 'h', 'a', 'i', 'n'})
-	err := InsertYield(testYield, "testDB.dat", 35, contractHash, 1)
+//test
+func TestMakeClaimChange(t *testing.T) {
+	db := "testDB.db"
+	setUpDB(db)
+	defer tearDown(db)
+
+	amt := uint64(500)
+	pubKey := generatePubKey()
+	yield := MakeYield(&pubKey, amt)
+	contractHash := block.HashSHA256([]byte("blokchain"))
+	blockHeight := uint64(35)
+	yieldIndex := uint16(1)
+
+	err := InsertYield(yield, db, blockHeight, contractHash, yieldIndex)
 	if err != nil {
-		t.Errorf("Failed to insert yield")
+		t.Errorf("Failed to insert yield: %v", err)
 	}
-	testClaim, err := MakeClaim("testDB.dat", testPubKey, 10000000)
-	if err != nil {
-		t.Errorf("Failed to claim yield")
+
+	testClaim, err := MakeClaim(db, pubKey, amt-5)
+	if err == nil {
+		t.Errorf("No change error struct returned")
+	} else {
+		if change, ok := err.(ChangeError); ok {
+			if change.Change != uint64(5) {
+				t.Errorf("Wrong change value, expected: 5, actual: %d", change)
+			}
+		} else {
+			t.Errorf("Different error given: %v", err)
+		}
 	}
+
 	if !bytes.Equal(testClaim.PreviousContractHash, contractHash) {
 		t.Errorf("Contract hashes do not match")
 	}
-	if testClaim.BlockIndex != 35 {
+	if testClaim.BlockIndex != blockHeight {
 		t.Errorf("Block indeces do not match")
 	}
-	if testClaim.YieldIndex != 1 {
+	if testClaim.YieldIndex != yieldIndex {
 		t.Errorf("Yield indeces do not match")
 	}
-	if !cmp.Equal(testClaim.PublicKey, testPubkey) {
+	if !reflect.DeepEqual(testClaim.PublicKey, pubKey) {
```

Comment:
Does this not work with `cmp`?

Reply:
No, apparently cmp does not support unexported fields

Reply:
https://godoc.org/github.com/google/go-cmp/cmp#AllowUnexported